### PR TITLE
BCDA-638 Documentation: add description of credentials

### PR DESCRIPTION
**Pull request by dhgreene**
### Fixes [BCDA-638](https://jira.cms.gov/browse/BCDA-638)
Up to this point we've been distributing long-lived access tokens.  Our auth providers are now ready to use credentials to create short-lived access tokens via `/auth/token`.  The goal of this PR is to document that change in the user guide, encryption walkthrough, and anywhere else needed.

### Proposed changes:
* Publish client ID and client secret instead of a token
* Document new auth steps, including screenshots
* Change terminology (`s/token/credentials/`) where needed

### Change Details
* Inserting screenshots in the encryption walkthrough required re-numbering.  To avoid potential cache issues, the screenshot naming scheme was also changed.
* A typo in the `pip install` instruction was corrected
* Some additional examples were added to the troubleshooting section

### Security Implications
There is a minor change in security: we're publishing credentials for `opensbx` instead of a token.  While no additional access is granted, the credentials allow for a new action (token creation).  This is a planned change, and brings us up to the point we had originally targeted for the release of `opensbx`.

### Acceptance Validation
Screen Shot 2019-04-19 at 2 49 56 PM
Screen Shot 2019-04-19 at 2 50 29 PM
Screen Shot 2019-04-19 at 2 50 41 PM

### Feedback Requested
* Typos, inaccuracies
* Visual distractions (for instance, line wrapping on the client secret and Basic authentication header would be ideal)
* Anything else!
* Note that @epfrankle will be reviewing the edits next sprint

### CODE REVIEW

File: bcda-site-static/decryption_walkthrough.md [hunk 1]
```diff
@@ -40,21 +41,42 @@ You will be interacting with the [BCDA API](./user_guide.html) in your browser u
 - Open your browser and navigate to:
 
 ### [https://sandbox.bcda.cms.gov/api/v1/swagger](./api/v1/swagger/){:target="_blank"}
-<img src="assets/img/decrypt_demo_01.png" alt="Screenshot of swagger" width="400" />
+<img src="assets/img/decrypt_walkthrough_01.png" alt="Screenshot of swagger" width="400" />
 
-Provide your token to get full access to the API.
+## Getting a token
 
+Use your credentials to get an access token.
```
Comment (knollfear) [inline bcda-site-static/decryption_walkthrough.md]: Should this link to the public sandbox ones?  not sure it should, but worth considering.

File: bcda-site-static/encryption.md [hunk 2]
```diff
@@ -79,7 +79,7 @@ The `KeyMap` object within our job status response has keys, values: `"<filename
 
 When you receive the final job status response, you should save the keys associated with the files so that they are available to you when you are ready to decrypt the file(s). You should also save the `output.url` and the `error.url`.
 
-When you are ready to decrypt the files, you make a request to `output.url` for the data file, and to `error.url` for the error file. These are protected endpoints, so you must obtain and use a token.
+When you are ready to decrypt the files, you make a request to `output.url` for the data file, and to `error.url` for the error file. These are protected endpoints, so you must use your credentials to obtain and use a token.
```
Comment (knollfear) [inline bcda-site-static/encryption.md]: so you must use your credentials to obtain and use a token.

so you must obtain and use a token.  The process for obtaining a token is describe here<link to Getting a token>

File: bcda-site-static/decryption_walkthrough.md (hunk 1 above)
Comment (dhgreene) [reply bcda-site-static/decryption_walkthrough.md]: Maybe.  How about:
`Use your credentials to get an access token.  If you need credentials, use `[these](https://sandbox.bcda.cms.gov/user_guide.html#authentication-and-authorization).

File: bcda-site-static/user_guide.md [hunk 3]
```diff
@@ -44,13 +44,34 @@ Not familiar with APIs? Here are some great introductions:
...
 
-{%- capture code -%}
-eyJhbGciOiJSUzUxMiIsInR5cCI6IkpXVCJ9.eyJhY28iOiJkM2U1MGEwNC0zMzA0LTRkMDEtYWUwMC1jNGNlOTIzODBkMTEiLCJleHAiOjE2MTYwODk3NTYsImlhdCI6MTU1MzAxNzc1NiwiaWQiOiJlNzFmMDQyYi1hMWZiLTQ0MDItYTgzZS1lZDlhYThlMTg4NmEiLCJzdWIiOiI1MDlkMjYwMy0wNzhiLTQ2YzgtODVmYi1lYzU3ZWEyY2QzYmQifQ.W2YKCNQNHkUukW1gP76cXr3J0JVeuYXSbgZQ9pAf2Rb_dnJ_GRn8g7rGwoGZUkCCv9fEOGUrbDpjpPJbNJUiwUqcfXCWkdQxTEfimAx5orC6UiNorjqPuKmloALWmN-B7b_62-BJ62u0xR5glbHl7CV5buI9yzWVzbMgvwuUH3VY3B7FQ-MXL3aqLtNoqlmfXjnARb4PsFBBReyJseIPpIbCQB3fUfV3DFL6wRWk4AW_Sa1w4UamzCyZER398cXE9CvpTylyVVdZSoP_p3V7tVyi5xVC8Jjf_zY3wJi2nc0ONqLqyMET8vfItVdYmJ6R4ShdIVRzDHFln7mXYVwOVw
-{%- endcapture -%}
+To get a token that can be used with protected endpoints, `POST` the following credentials using [Basic Authentication](https://en.wikipedia.org/wiki/Basic_access_authentication) to `/auth/token`:
+
+* Client ID: `09869a7f-46ce-4908-a914-6129d080a2ae`
+* Client Secret: `64916fe96f71adc79c5735e49f4e72f18ff941d0dd62cf43ee1ae0857e204f173ba10e4250c12c48`
+
```
Comment (knollfear) [inline bcda-site-static/user_guide.md]: * Basic Authentication *: W/Elkfjdflkjdflkbjdflkfdjfdlkjfdlkfgdjfdglkjfgdlkjfgdlkjdfg

*    *line 61

File: bcda-site-static/decryption_walkthrough.md [hunk 4]
```diff
@@ -40,21 +41,42 @@ You will be interacting with the [BCDA API](./user_guide.html) in your browser u
...
+
+- Click the `/auth/token` link
+
+<img src="assets/img/decrypt_walkthrough_03.png" alt="Token endpoint" width="600" />
+
+- Click `Try it out`, then `Execute`
+
+<img src="assets/img/decrypt_walkthrough_04.png" alt="Getting a token" width="600" />
+
+If you're successful, your token will appear as pictured in the `response body`.  You can now present this token to get full access to the API.
```
[the remark this answers is not in the document]
Comment (dhgreene) [reply bcda-site-static/decryption_walkthrough.md]: Yes.  You prefer use?  Done (well, will be in next commit).  :smile:

File: bcda-site-static/decryption_walkthrough.md [hunk 5]
```diff
@@ -63,23 +85,23 @@ There are two types of encrypted files that can be downloaded: Patient and Expla
...
-<img src="assets/img/decrypt_demo_06.png" alt="Requesting job status" width="600" />
+<img src="assets/img/decrypt_walkthrough_09.png" alt="Requesting job status" width="600" />
 
 - Type the job ID you received
 - Click `Execute`
 
-<img src="assets/img/decrypt_demo_07.png" alt="Job status response" width="600" />
+<img src="assets/img/decrypt_walkthrough_10.png" alt="Job status response" width="600" />
 
 1. Depending on the size of the file, the job may take some time.  If the job is not yet complete, status information will be shown.  Simply wait a few seconds and click execute again until the job completes. You will then get a result for a completed job as shown below.  You can download the file from the URL provided.
```
[the remark this answers is not in the document]
Comment (dhgreene) [reply bcda-site-static/decryption_walkthrough.md]: Screen Shot 2019-04-19 at 12 44 17 PM

File: bcda-site-static/decryption_walkthrough.md [hunk 6]
```diff
@@ -121,20 +143,23 @@ You are now ready to decrypt the file!  Your sample decryption tool will print t
...
 * [API documentation](./user_guide.html)
 
 ## Troubleshooting
 ### Authentication problems in Swagger
-- Has your token expired?  [Ask the support team](https://groups.google.com/forum/#!forum/bc-api){:target="_blank"} for a new token.
-- Is it possible you clicked on `Logout`?  Is the lock on the `Authorize` icon not closed?  Click it again, and after pasting your token in the `api_key` box, make sure to click the `Authorize` button.
+- Did you use the credentials (Client ID and Client Secret) from the [user guide](./user_guide.html) with Basic authorization?
+- After entering your credentials, did you get an access token from `/auth/token`?
+- Did you put your access token in the `bearer_token` section of the authorization dialog?
+- Has your token expired?  Use your credentials to get a new token from `/auth/token`.
```
[the remark this answers is not in the document]
Comment (dhgreene) [reply bcda-site-static/decryption_walkthrough.md]: It's configurable.  In production, it should be 5 minutes.  We may choose to extend that to 60 minutes in `opensbx`.

[the remark this answers is not in the document]
Comment (knollfear) [reply bcda-site-static/decryption_walkthrough.md]: Whatever it is, people will feel better if they know.

[the remark this answers is not in the document]
Comment (dhgreene) [reply bcda-site-static/decryption_walkthrough.md]: Working on getting a decision from the team on this.

**Comment by dhgreene** [comment]
Screenshots in PR description updated

File: bcda-site-static/decryption_walkthrough.md [hunk 7]
```diff
@@ -40,21 +41,42 @@ You will be interacting with the [BCDA API](./user_guide.html) in your browser u
...
-<img src="assets/img/decrypt_demo_02.png" alt="Authorization dialog" width="350" />
+<img src="assets/img/decrypt_walkthrough_05.png" alt="Authorization dialog" width="350" />
 
 - Click `Authorize`, then `Close`
 
-<img src="assets/img/decrypt_demo_03.png" alt="API menu" width="600" />
+<img src="assets/img/decrypt_walkthrough_06.png" alt="API menu" width="600" />
 
 ## Requesting a file
 There are two types of encrypted files that can be downloaded: Patient and Explanation of Benefit (EoB). The example below id is for downloading the EoB.
```
Comment (knollfear) [inline bcda-site-static/decryption_walkthrough.md]: out of scope for this PR, but is this 3 now that coverage is enabled?

File: bcda-site-static/user_guide.md [hunk 8]
```diff
@@ -44,13 +44,50 @@ Not familiar with APIs? Here are some great introductions:
...
 
-{%- capture code -%}
-eyJhbGciOiJSUzUxMiIsInR5cCI6IkpXVCJ9.eyJhY28iOiJkM2U1MGEwNC0zMzA0LTRkMDEtYWUwMC1jNGNlOTIzODBkMTEiLCJleHAiOjE2MTYwODk3NTYsImlhdCI6MTU1MzAxNzc1NiwiaWQiOiJlNzFmMDQyYi1hMWZiLTQ0MDItYTgzZS1lZDlhYThlMTg4NmEiLCJzdWIiOiI1MDlkMjYwMy0wNzhiLTQ2YzgtODVmYi1lYzU3ZWEyY2QzYmQifQ.W2YKCNQNHkUukW1gP76cXr3J0JVeuYXSbgZQ9pAf2Rb_dnJ_GRn8g7rGwoGZUkCCv9fEOGUrbDpjpPJbNJUiwUqcfXCWkdQxTEfimAx5orC6UiNorjqPuKmloALWmN-B7b_62-BJ62u0xR5glbHl7CV5buI9yzWVzbMgvwuUH3VY3B7FQ-MXL3aqLtNoqlmfXjnARb4PsFBBReyJseIPpIbCQB3fUfV3DFL6wRWk4AW_Sa1w4UamzCyZER398cXE9CvpTylyVVdZSoP_p3V7tVyi5xVC8Jjf_zY3wJi2nc0ONqLqyMET8vfItVdYmJ6R4ShdIVRzDHFln7mXYVwOVw
+To get a token that can be used with protected endpoints, `POST` the following credentials using [Basic Authentication](https://en.wikipedia.org/wiki/Basic_access_authentication) to `/auth/token`:
```
Comment (knollfear) [inline bcda-site-static/user_guide.md]: This link needs to open in a new window please.

**Pull request merged**